billiford/go-clouddriver
# Adds support for git/repo artifacts

Adds support for the `git/repo` artifact type

To test locally

1. Create artifact account config file `/opt/spinnaker/artifacts/config/git-repo.json`
```json
{
  "baseURL": "https://github.com",
  "name": "git-repo",
  "types": [
    "git/repo"
  ]
}
```
2. Start `go-clouddriver`
```
make run
```
3. Make request
```
curl "http://localhost:7002/artifacts/fetch" \
     -s -L -X PUT -H 'Content-Type: application/json' -d '{
	   "version": "master",
	   "artifactAccount": "git-repo",
	   "reference": "https://github.com/billiford/go-clouddriver",
	   "type": "git/repo"
	 }' | tar t -
```

Should see results like
```
.github/
.github/workflows/
.github/workflows/go.yml
.gitignore
Makefile
README.md
api/
api/swagger.yaml
cmd/
cmd/clouddriver/
cmd/clouddriver/clouddriver.go
docker/
docker/Dockerfile
docker/build.sh
...
pkg/sql/sqlfakes/
pkg/sql/sqlfakes/fake_client.go
pkg/task.go
```
